--- src/ushare.c	2014-06-06 11:01:22.731541612 +0200
+++ src/ushare.c	2014-06-06 11:06:20.367529164 +0200
@@ -188,7 +188,7 @@
   if (strcmp (request->DevUDN + 5, ut->udn))
     return;
 
-  ip = request->CtrlPtIPAddr.s_addr;
+  ip = ((struct in_addr *)&(request->CtrlPtIPAddr))->s_addr;
   ip = ntohl (ip);
   sprintf (val, "%d.%d.%d.%d",
            (ip >> 24) & 0xFF, (ip >> 16) & 0xFF, (ip >> 8) & 0xFF, ip & 0xFF);
